opaque_keys/__init__.py

```diff
@@ -183,6 +208,16 @@ def from_string(cls, serialized):
         Args:
             serialized: A stringified form of a :class:`OpaqueKey`
         """
+        # OpaqueKeys are immutable, so we can share them. Sharing the keys gives
+        # two benefits:
+        #    1. Remove parsing/construction costs.
+        #    2. Remove the costs to calculate the hash() and to_unicode() which
+        #       are cached on the individual OpaqueKey
+        #    3. It short circuits True if they're pointing at the same object,
+        #       which is a cheap comparison (and frequent).
```

Comment:
A re-wording might help this comment, like:
`Inexpensive comparsion if pointing at the same object (which is a frequent occurrence).`